hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -223,11 +224,27 @@ usePropertyLsp kn pId p = do
 
 -- ---------------------------------------------------------------------
 
-extractRange :: Range -> T.Text -> T.Text
-extractRange (Range (Position sl _) (Position el _)) s = newS
+-- | Extracts lines in the text overlapping the given range.
+--
+-- See also: 'extractTextInRange'
+extractOverlappingLinesWithRange :: Range -> T.Text -> T.Text
+extractOverlappingLinesWithRange (Range (Position sl _) (Position el _)) s = newS
```

Comment:
I am not sure if I like the name, but I can't think of anything else. @michaelpj any ideas?

Reply:
If it were me, I would be tempted to instead provide `expandToLines :: Range -> Range` and make `extractTextInRange` exact.

Reply:
Perhaps `extractLinesInRange`?

Reply:
Migrating to Rope would be nice, since we would presumably speed up the formatting code actions for selections :)

Reply:
It's not necessarily just fine. This is in a code action, and I don't think we have resolve support here. So this may make the code action request grindingly slow on large files on 9.4 if there is an out-of-scope identifier. Maybe we're okay with that but it's potentially a real regression. I'll make a ticket anyway.

Reply:
https://github.com/haskell/haskell-language-server/issues/3720

Reply:
Fair enough.

Reply:
I followed the suggestion from @michaelpj in b2c5f78513b65b1f0efe2936f1a950b993870450.
Migration to `Rope` (#3720) sounds promising. It seems like a change affecting wider range than the initial purpose of this PR, so I would rather leave it to the future refactoring (by someone else, or myself if the time allows :grin:).

Reply:
Yes, definitely out of scope for this PR!